frontend-ssr/pages/submissions.tsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import Head from 'next/head';
+import Footer from '~/components/Footer';
+import Header from '~/components/Header';
+
+import ExerciseSubmission from '../components/exercise-submissions/ExerciseSubmission';
+import ExerciseService from '../services/ExerciseSubmissions.service';
+import {
+  loadSubmissions,
+  searchSubmissions,
+} from '~/redux/exercise-submissions/exercise-submissions.actions';
+import ExerciseSubmissionSkeleton from '~/components/exercise-submissions/ExerciseSubmissionSkeleton';
+import SweetAlertService from '../services/sweet-alert/SweetAlert.service';
+import Search from '~/components/Search';
+import { RootState } from '../redux/root.reducer';
+
+import styles from './submissions.module.scss';
+
+interface State {
+  loading: boolean;
+  prevY: number;
+}
+
+class Submissions extends React.Component<ConnectedProps<typeof connector>, State> {
+  private observer: IntersectionObserver;
+
+  private buttonRef = React.createRef<HTMLButtonElement>();
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: props.submissions.submissions === undefined,
+      prevY: 0,
+    };
+  }
+
+  componentDidMount() {
+    let { submissions } = this.props;
+
+    if (submissions.submissions === undefined) {
+      this.getData();
+    }
+
+    const options = {
+      threshold: 0.7,
+    };
+
+    this.observer = new IntersectionObserver(this.loadMore, options);
+    this.observeNextPage();
```

Comment:
I don't think we need a separate function for this. Instead I would create an: `initIntersectionObserver` function with the `options` object and the initialization code. It would make `componentDidMount` cleaner.

Reply:
Also, let's not observe the button, it happens to be at the end but that's not always a guarantee.

Instead I would add an "invisible" element at the end which we'll observe for intersections.

Reply:
Another point here is that we should only initialize the observer if we haven't reached the end. See another comment down  below regarding that.